diff --git a/build/config/sysroot.gni b/build/config/sysroot.gni
index fb0f3c3..69bbf7e 100644
--- a/build/config/sysroot.gni
+++ b/build/config/sysroot.gni
@@ -30,14 +30,16 @@ if (current_toolchain == default_toolchain && target_sysroot != "") {
   if (current_cpu == "mipsel") {
     sysroot = rebase_path("//mipsel-sysroot/sysroot")
   } else if (use_default_linux_sysroot && !is_fuchsia) {
-    if (current_cpu == "x64") {
+    if (target_cpu == "x64") {
       sysroot = rebase_path("//build/linux/debian_sid_amd64-sysroot")
-    } else {
+    } else if (target_cpu == "arm64") {
       sysroot = rebase_path("//build/linux/debian_sid_arm64-sysroot")
+    } else {
+        sysroot = ""
     }
-    assert(
-        exec_script("//build/dir_exists.py", [ sysroot ], "string") == "True",
-        "Missing sysroot ($sysroot). To fix, run: build/linux/sysroot_scripts/install-sysroot.py --arch=$current_cpu")
+    #assert(
+    #    exec_script("//build/dir_exists.py", [ sysroot ], "string") == "True",
+    #    "Missing sysroot ($sysroot). To fix, run: build/linux/sysroot_scripts/install-sysroot.py --arch=$current_cpu")
   } else {
     sysroot = ""
   }
